### PR TITLE
feat: add check before syncing doc

### LINKED ISSUE #20470: [Syncing] Improve user awareness of requirements to initiate syncing
# Feature

This feature is part of our ongoing efforts to simplify the on-boarding funnel.

### User Story

- As a user
- I want to be clearly informed of what actions I need to take to sync my devices
- So that I can start the sync flow and have a frictionless experience

### Description

There are two flows to consider. One in the [onboarding](https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=12958-36180&t=L2nE793eZzGGElja-0), and one when the user is [already logged in](https://www.figma.com/design/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=3158-5671&t=GIFj5fjqucAlUonV-0).

We are betting that adding a little bit of friction early in the sync flow to prevent users from facing obvious errors and thus dropping off is advantageous to users.

**In scope** is only the steps to show checkboxes. The moment the user is able to press on "Continue", it should trigger the same flows we already use to start syncing.

### Resources

- Figma flow https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=12958-36180&t=L2nE793eZzGGElja-0

### PR DESCRIPTION
**Pull request by yqrashawn**
fixes #20470

### Summary

This PR relies on new onboarding flow in PR #20386
Will be ready for review once that PR merged

### Review notes
add `status-im.common.check-before-syncing.view`
`status-im.common.check-before-syncing.view/view` can be used by other code in combination with `:show-bottom-sheet`

### Testing notes
flow changed, affects e2e tests

#### Platforms

- Android
- iOS

#### Areas that maybe impacted
- sign in by syncing after intro
- sign in by syncing after logged out
- add syncing device after logged in

status: ready

### CODE REVIEW

**Comment by yqrashawn** [comment]
https://github.com/status-im/status-mobile/assets/15090582/969aba67-5461-4592-a86b-2ee8a9c3c09a


https://github.com/status-im/status-mobile/assets/15090582/fa3d459a-54a0-4d92-8f09-b100a86bbe82

File: src/status_im/common/check_before_syncing/style.cljs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+(def buttons-container
+  {:flex-direction   :row
+   :justify-content  :space-between
+   :gap              12
+   :padding-vertical 12})
+
+(def button {:flex 1})
+
+(def checkbox {:margin-top 16 :flex-direction :row :align-items :center})
+(def checkbox-text {:margin-left 10})
```
Comment (Parveshdhull) [inline src/status_im/common/check_before_syncing/style.cljs]: Thank you @yqrashawn for your PR.
Please format these lines. (empty lines and map)

File: src/status_im/common/check_before_syncing/view.cljs [hunk 2]
```diff
@@ -0,0 +1,64 @@
+(ns status-im.common.check-before-syncing.view
+  (:require
+    [quo.core :as quo]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [status-im.common.check-before-syncing.style :as style]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+
```
Comment (Parveshdhull) [inline src/status_im/common/check_before_syncing/view.cljs]: remove empty line

File: src/status_im/common/check_before_syncing/view.cljs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  [rn/view {:style style/checkbox}
+   [quo/selectors
+    {:type      :checkbox
+     :blur?     true
+     :on-change on-change}]
+   [quo/text {:style style/checkbox-text} description]])
+
+(defn view
+  []
+  (let [all-checked?            (reagent/atom 0)
```
Comment (Parveshdhull) [inline src/status_im/common/check_before_syncing/view.cljs]: Shouldn't we use rn/use-state instead?

File: translations/en.json [hunk 4]
```diff
@@ -179,6 +179,11 @@
     "chat-settings": "Chat settings",
     "chat-with-friends": "Chat privately with friends",
     "chats": "Chats",
+    "check-before-syncing":"Check before syncing",
```
Comment (Parveshdhull) [inline translations/en.json]: Other translations also have empty space after `:`, maybe we should add for consistency.

File: src/status_im/common/check_before_syncing/view.cljs (hunk 3 above)
Comment (Parveshdhull) [reply src/status_im/common/check_before_syncing/view.cljs]: why all-checked? is a number?

File: src/status_im/common/check_before_syncing/view.cljs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        checkbox-with-on-change (partial checkbox
+                                         #(if %
+                                            (swap! all-checked? inc)
+                                            (swap! all-checked? dec)))]
+    (fn [{:keys [on-continue]}]
```
Comment (Parveshdhull) [inline src/status_im/common/check_before_syncing/view.cljs]: I think we will not need this.

**Comment by yqrashawn** [comment]
Thanks for the review @Parveshdhull 
I've addressed all the above reviews

File: translations/en.json [hunk 6]
```diff
@@ -179,6 +179,11 @@
     "chat-settings": "Chat settings",
     "chat-with-friends": "Chat privately with friends",
     "chats": "Chats",
+    "check-before-syncing": "Check before syncing",
+    "check-before-syncing-doc-description":"To sync your devices successfully, make sure to check and complete these steps:",
+    "check-before-syncing-doc-checkbox-1":"Both devices are on the same network",
+    "check-before-syncing-doc-checkbox-2":"Make sure you are logged in on the other device",
+    "check-before-syncing-doc-checkbox-3":"Disable the firewall on your device",
```
Comment (Parveshdhull) [inline translations/en.json]: spaces are still missing at other occurrences.

**Review by Parveshdhull — APPROVED** [review]

File: translations/en.json (hunk 6 above)
Comment (yqrashawn) [reply translations/en.json]: Uh sorry for that, I forgot that I added so many translations in this PR. I'll fix them in this PR.

File: src/status_im/common/check_before_syncing/view.cljs [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    {:type      :checkbox
+     :blur?     true
+     :on-change on-change}]
+   [quo/text {:style style/checkbox-text} description]])
+
+(defn- compute-new-checked-count
+  [set-checked-count checked?]
+  (if checked?
+    (set-checked-count #(inc %))
+    (set-checked-count #(dec %))))
```
Comment (clauxx) [inline src/status_im/common/check_before_syncing/view.cljs]: could just use `(set-checked-count inc)` and same for `dec`?

File: src/status_im/common/check_before_syncing/view.cljs [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+(defn- hide-bottom-sheet
+  []
+  (rf/dispatch [:hide-bottom-sheet]))
+
+(defn view
+  [{:keys [on-continue]}]
```
Comment (clauxx) [inline src/status_im/common/check_before_syncing/view.cljs]: could rename this to `on-submit` to avoid having `real-on-continue`

**Review by clauxx — APPROVED** [review]
Looks good!

File: src/status_im/common/check_before_syncing/view.cljs (hunk 7 above)
Comment (yqrashawn) [reply src/status_im/common/check_before_syncing/view.cljs]: fixed, I thought there's another arg after prevState and turns out there's not

**Comment by yqrashawn** [comment]
Thanks for the review @Parveshdhull @clauxx, new commit addressed above review comments

**Comment by qoqobolo** [comment]
Thanks for your work @yqrashawn, everything looks good to me, but I have one question:

is the checkbox/button color in the scope of this PR?
Currently, they don't inherit the custom color. WDYT, should it be fixed here? 
 
Screenshot 2024-06-28 at 14 06 56

**Comment by qoqobolo** [comment]
@yqrashawn Alisher has just updated the design (now VPN is mentioned in the copy). Discussed [here](https://github.com/status-im/status-mobile/issues/20525#issuecomment-2196906128) and [here](https://discord.com/channels/624634427930312714/928625369542713396/1256238342966743152) 

Figma: https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=12958-36180&t=Lk6h5r7RglW1XPcc-4

Would you mind updating the copy, please?

**Comment by yqrashawn** [comment]
> copy change

I've updated the copy, thanks for the link. @qoqobolo 

> customization color

Fixed the one in settings -> syncing (only one after login)
CleanShot 2024-07-01 at 09 49 07

**Comment by qoqobolo** [comment]
Thank you @yqrashawn! LGTM

@Francesca-G the PR is ready for design review, thanks!

**Review by Francesca-G — COMMENTED** [review]
Here's the [review](https://www.figma.com/design/oRQ5TcgMHKkK79EUfURg3e/Review-Onboarding-%2F-Mobile?node-id=33-1278&t=m7v5LaNMgmEixvim-1) 🙏

**Comment by yqrashawn** [comment]
Thanks for the review @Francesca-G, styles are fixed now


simulator_screenshot_94E7C589-24C5-4C6D-97BE-26FDEE08B533

**Review by Francesca-G — APPROVED** [review]

**Pull request merged**